plugins/nanospell/plugin.js

```diff
@@ -664,19 +664,17 @@
 				var blockList = event.data.blockList;
 				var url = resolveAjaxHandler();
 				var callback = function (data) {
-					var bookmarks;
+					var selectionStart = editor.getSelection().getStartElement();
+					var rootElement;
 					parseRpc(data, words);
-					editor.lockSelection();
-					bookmarks = editor.getSelection().createBookmarks(true);
+
 					for (var i = 0; i < blockList.length; i++) {
-						var rootElement = blockList[i];
+						rootElement = blockList[i];
 						editor.fire(EVENT_NAMES.START_RENDER, {
 							root: rootElement,
-							needsBookmarkCreated: false,
+							needsBookmarkCreated: selectionStart ? rootElement.contains(selectionStart) || rootElement.equals(selectionStart) : null,
```

Comment:
I hadn't considered that future code plans on changing how this works. In generally, avoiding doing work in a loop that can be done outside it is good, but it sounds like it will introduce future complexity.